query-creator/create-query/steps/aws.js

```diff
@@ -101,16 +101,20 @@ exports.generateSQL = (queryId, runnable, resource, denormalizedResource) => {
     }
   })
 
-  return `WITH activities AS ( SELECT * FROM "report-service"."activity_structure" WHERE structure_id = '${queryId}' )
+  return `WITH activities AS ( SELECT *, cardinality(questions) as num_questions FROM "report-service"."activity_structure" WHERE structure_id = '${queryId}' )
 
 SELECT
-  ${["null as remote_endpoint"].concat(selectColumnPrompts).join(",\n  ")}
+  ${["null as remote_endpoint,\n  null as num_questions,\n  null as num_answers,\n  null as percent_complete"].concat(selectColumnPrompts).join(",\n  ")}
 FROM activities
 
 UNION ALL
 
 SELECT
-  ${["remote_endpoint"].concat(selectColumns).join(",\n  ")}
+  ${[`remote_endpoint,
+  activities.num_questions,
+  cardinality(array_intersect(map_keys(kv1),map_keys(activities.questions))) as num_answers,
+  round(100.0 * cardinality(array_intersect(map_keys(kv1),map_keys(activities.questions))) / activities.num_questions, 1) as percent_complete`
+    ].concat(selectColumns).join(",\n  ")}
```

Comment:
It'd be useful to restructure this so it is easier to comment and document what is happening.
Perhaps you could define the completionColumns in a variable above and then do something like:
```
SELECT
  ${[
    "remote_endpoint",
     ...completionColumns,
     ...selectColumns
  ].join(",\n ")}

Reply:
Sounds good.  I pushed a commit to address this.